src/shared/Microsoft.Git.CredentialManager/GitVersion.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Git.CredentialManager
+{
+    public class GitVersion : IComparable, IComparable<GitVersion>
```

Comment:
Implementing `IComparable` on types can be hard to get right; looks like you've done a great job - nice! 😁 

Did you consider also implementing the `IEquatable` and `IEquatable<GitVersion>` interfaces?
The implementation is already there in your overridden `bool Equals(object)` method but adding the interface will mean should we ever put `GitVersion` in a dictionary or set, that collection can _know_ it can be equated.

This change does not block this PR by the way..